meta-client/src/main/java/org/apache/helix/metaclient/impl/zk/ZkMetaClient.java

```diff
@@ -67,14 +67,33 @@ public ZkMetaClient(ZkMetaClientConfig config) {
   }
 
   @Override
-  public void create(String key, T data) {
-    // TODO: This function is implemented only for test. It does not have proper error handling
-    _zkClient.create(key, data, ZooDefs.Ids.OPEN_ACL_UNSAFE, CreateMode.PERSISTENT);
+  public void create(String key, Object data) {
+    try {
+      create(key, data, EntryMode.PERSISTENT);
+    } catch (Exception e) {
+      throw new MetaClientException(e);
+    }
   }
 
   @Override
-  public void create(String key, T data, EntryMode mode) {
+  public void create(String key, Object data, MetaClientInterface.EntryMode mode) {
 
+    try{
+      _zkClient.create(key, data, metaClientModeToZkMode(mode));
+    } catch (ZkException | KeeperException e) {
+      throw new MetaClientException(e);
+    }
+  }
+
+  private CreateMode metaClientModeToZkMode(EntryMode mode) throws KeeperException {
+    int zkFlag = -1;
+    if (mode.name().equals(EntryMode.PERSISTENT.name())) {
+      zkFlag = 0;
+    }
+    if (mode.name().equals(EntryMode.EPHEMERAL.name())) {
+      zkFlag = 1;
```

Comment:
can we just return the mode here? Including a separate int value could be confusing.